src/mds/Server.cc

```diff
@@ -1085,9 +1085,13 @@ void Server::recover_filelocks(CInode *in, bufferlist locks, int64_t client)
  */
 void Server::recall_client_state(float ratio)
 {
-  int max_caps_per_client = (int)(g_conf->mds_cache_size * .8);
-  int min_caps_per_client = 100;
-
+  int max_caps_per_client = (int)(g_conf->mds_cache_size * g_conf->max_ratio_caps_per_client);
+  int min_caps_per_client = (int)(g_conf->min_caps_per_client);
+  if (max_caps_per_client < min_caps_per_client) {
+    max_caps_per_client = min_caps_per_client + 1;
```

Comment:
This breaks a test: http://pulpito.ceph.com/pdonnell-2017-07-19_19:11:33-fs-wip-pdonnell-testing-20170718-distro-basic-smithi/1420077/

Key line is:

`2017-07-19T23:56:31.006 DEBUG:tasks.ceph_test_case:wait_until_equal: 202 != 80, waiting...
2017-07-19T23:56:33.637 INFO:tasks.ceph.mds.a.smithi031.stderr:2017-07-19 23:56:33.571051 7f4b5aab6700 -1 mds.0.server max_caps_per_client 101<min_caps_per_client 100`

The max_caps_per_client should be `80` which the test relies on. Please update the test to also change the min caps per client:

https://github.com/ceph/ceph/blob/359cd17ee98b73559d07c04a030a9506ab244000/qa/tasks/cephfs/test_client_limits.py#L45